File: src/main/java/com/dxschool/lightme/caseuser/controller/dto/CaseUserDetailResponse.java
package com.dxschool.lightme.caseuser.controller.dto;

import com.dxschool.lightme.schedule.controller.dto.ArtistScheduleResponse;

import java.util.List;

public record CaseUserDetailResponse(
        CaseUserResponse caseUser,
        List<ImageResponse> images,
        List<VideoResponse> videos,
        List<MusicResponse> playlist,
        ArtistScheduleResponse upcomingSchedule,
        int dDay
) {
    public static CaseUserDetailResponse of(CaseUserResponse caseUser,
                                            List<ImageResponse> images,
                                            List<VideoResponse> videos,
                                            List<MusicResponse> playlist,
                                            ArtistScheduleResponse upcomingSchedule,
                                            int dDay
    ) {
        return new CaseUserDetailResponse(caseUser, images, videos, playlist, upcomingSchedule, dDay);
    }
}

File: android/app/src/main/java/com/example/tfisland/model/dto/CaseUserDetailResponse.kt
package com.example.tfisland.model.dto

data class CaseUserDetailResponse(
    val caseUserResponse: CaseUserResponse,
    val imageResponseList: List<ImageResponse>,
    val videoResponseList: List<VideoResponse>,
    val musicResponseList: List<MusicResponse>,
    val artistScheduleReponse: ArtistScheduleResponse,
    val dDay: Int
)
